internal/controller/controllers/infraenv_controller_test.go

```diff
@@ -134,6 +135,53 @@ var _ = Describe("infraEnv reconcile", func() {
 		Expect(result).To(Equal(ctrl.Result{}))
 	})
 
+	Context("Mirror registry", func() {
+
+		createUserMirrorRegistryConfigmap := func(registryToml string, certificate string) {
+			registryCM := newUserProvidedRegistryCM(registryToml, certificate)
+			Expect(c.Create(ctx, registryCM)).ShouldNot(HaveOccurred())
+		}
+
+		When("the user-provided valid image registry contains correct data", func() {
+			It("Successfully parse and set the given user configmap", func() {
```

Comment:
Since this is all just one big test, it can just be one big test with much less indentation.